x-pack/plugins/security_solution/server/endpoint/routes/metadata/index.ts

```diff
@@ -4,51 +4,18 @@
  * you may not use this file except in compliance with the Elastic License.
  */
 
-import { IRouter, Logger, RequestHandlerContext } from 'kibana/server';
-import { SearchResponse } from 'elasticsearch';
+import { IRouter } from 'kibana/server';
 import { schema } from '@kbn/config-schema';
-import Boom from 'boom';
 
-import { metadataCurrentIndexPattern } from '../../../../common/endpoint/constants';
-import { getESQueryHostMetadataByID, kibanaRequestToMetadataListESQuery } from './query_builders';
-import {
-  HostInfo,
-  HostMetadata,
-  HostMetadataDetails,
-  HostResultList,
-  HostStatus,
-} from '../../../../common/endpoint/types';
+import { HostStatus, MetadataQueryStrategyVersions } from '../../../../common/endpoint/types';
 import { EndpointAppContext } from '../../types';
-import { AgentService } from '../../../../../ingest_manager/server';
-import { Agent, AgentStatus } from '../../../../../ingest_manager/common/types/models';
-import { findAllUnenrolledAgentIds } from './support/unenroll';
-import { findAgentIDsByStatus } from './support/agent_status';
+import { getLogger, getMetadataListRequestHandler, getMetadataRequestHandler } from './handlers';
 
-interface MetadataRequestContext {
-  agentService: AgentService;
-  logger: Logger;
-  requestHandlerContext: RequestHandlerContext;
-}
-
-const HOST_STATUS_MAPPING = new Map<AgentStatus, HostStatus>([
-  ['online', HostStatus.ONLINE],
-  ['offline', HostStatus.OFFLINE],
-  ['unenrolling', HostStatus.UNENROLLING],
-]);
-
-/**
- * 00000000-0000-0000-0000-000000000000 is initial Elastic Agent id sent by Endpoint before policy is configured
- * 11111111-1111-1111-1111-111111111111 is Elastic Agent id sent by Endpoint when policy does not contain an id
- */
-
-const IGNORED_ELASTIC_AGENT_IDS = [
-  '00000000-0000-0000-0000-000000000000',
-  '11111111-1111-1111-1111-111111111111',
-];
-
-const getLogger = (endpointAppContext: EndpointAppContext): Logger => {
-  return endpointAppContext.logFactory.get('metadata');
-};
+export const BASE_ENDPOINT_ROUTE = '/api/endpoint';
+export const METADATA_REQUEST_V1_ROUTE = `${BASE_ENDPOINT_ROUTE}/v1/metadata`;
+export const GET_METADATA_REQUEST_V1_ROUTE = `${METADATA_REQUEST_V1_ROUTE}/{id}`;
+export const METADATA_REQUEST_ROUTE = `${BASE_ENDPOINT_ROUTE}/metadata`;
+export const GET_METADATA_REQUEST_ROUTE = `${METADATA_REQUEST_ROUTE}/{id}`;
```

Comment:
Side note: it would be great if these two `const` could be elevated to `x-pack/plugins/security_solution/common/endpoint/constants.ts` so that the UI can also use it. See the entries there now for Trusted apps for a reference